### PR TITLE
Support CloudLinux Build System

### PR DESCRIPTION
**Pull request by SF-300**

### CODE REVIEW

File: buildsys-pre-build [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    finally:
+        with open(path, "w") as f:
+            f.write(content)
+
+
+def _main():
+    # NOTE: For reasons unknown, the Build System clones repository under 'mockbuild' user
+    #       but executes 'buildsys-pre-build' script as 'root'. As 'buildsys-pre-build' script
+    #       invokes Makefile, which in turn invokes 'git archive' command, the latter fails due
+    #       to 'dubious ownership' error. This hack sidesteps the problem but should be fixed on
```
Comment (helldotcat) [inline buildsys-pre-build:28]: Do some task/ussue/proposal for build system related to this issue exist?

File: buildsys-pre-build [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    #       the side of Build System in the long run.
+    subprocess.call("git config --global --add safe.directory /srv/pre_build".split())
+    # NOTE: CloudLinux Build System redefines some macros, including %_rpmfilename.
+    #       This makes an upstream Makefile target "sources" to fail as it expects that
+    #       RPMs are sorted into directories with names corresponding to architectures.
+    #       This patch makes the build system to temporary use the default value of %_rpmfilename
+    #       when Makefile is executed, while reverting it back for rpmbuild invocation from inside
+    #       the Build System.
+    with _rpmmacros_file_patched():
```
Comment (helldotcat) [inline buildsys-pre-build:37]: When build process is killed, contextmanager exit is not run, and we get invalid `/root/.rpmmacros` file? Looks not good.

Comment (SF-300) [reply buildsys-pre-build:37]: CLBS runs each build in its own chroot environment so won't affect any other build. As leapp-repository is build in two-phases, the patch is only needed for internal meta-RPMs build during the first phase (via srpm Makefile target) and must be reverted before main SRPM is build by CLBS itself during the second phase.

File: buildsys-pre-build (hunk 1 above)
Comment (SF-300) [reply buildsys-pre-build:28]: [Yes](https://cloudlinux.atlassian.net/browse/AL-5227?atlOrigin=eyJpIjoiOTAyYzYxMDYwNDI2NDNhYTk3MzVmZjJhNDJkMGViMjIiLCJwIjoiamlyYS1zbGFjay1pbnQifQ).

**Pull request merged**